### PR TITLE
Adds efficient estimation for BADGES_SOLD

### PR DESCRIPTION
**Pull request by RobRuana**
The `BADGES_SOLD` estimate is hidden behind a `badges_sold_estimate_enabled` feature flag. This pull request also removes `HARDCORE_OPTIMIZATIONS_ENABLED` in favor of the new estimate.

I am going to merge this now, because I need to deploy it on the load testing system. However, @kitsuta and @EliAndrewC, I would love a pull request review!

Please hit me up with any questions or concerns, and I'll be happy to revise or revert things 😄

### CODE REVIEW

**Comment by thaeli** [comment]
It looks like this can result in slightly "overselling" badges in a close to sellout situation. That's reasonable enough for when the actual-max-badge-count is a kinda squishy target (Super definitely qualifies) but we should be sure to clearly document exactly what circumstances and how much oversell is possible. (Or if this is display only, about how much error is possible and what's the distribution of that error - I _think_ it will always estimate low for badges sold?)

I am very happy to get rid of HARDCORE_OPTIMIZATIONS_ENABLED, that was kinda hacky. Almost like we wrote it while servers were going down under load or something..

**Comment by RobRuana** [comment]
This would actually lead to "underselling" badges. It estimates _high_ for badges sold. It counts ALL non-staff attendee rows as a sold badge. So any refunded, invalid, or placeholder badges are still considered a "sold" badge in the estimate. It will look like we've sold out before we actually sell out.

I plan on flipping this back to `False` as soon as we get past prereg.

**Comment by RobRuana** [comment]
Well... let me amend that. I suppose it depends on how accurate `pg_class.reltuples` is. I doubt it will ever report MORE rows than exist in the table, so under heavy load it will probably under-report the number of badges sold by a bit.

However, that may actually be offset by the number of refunded/invalid/placeholder badges, so... ¯\_(ツ)_/¯

File: uber/config.py [hunk 1]
```diff
@@ -222,28 +221,34 @@ def ATTENDEE_BADGES_SOLD(self):
...
-        with Session() as session:
-            attendees = session.query(Attendee)
-            individuals = attendees.filter(or_(
-                Attendee.paid == self.HAS_PAID,
-                Attendee.paid == self.REFUNDED)
-            ).filter(Attendee.badge_status == self.COMPLETED_STATUS).count()
+        if self.BADGES_SOLD_ESTIMATE_ENABLED:
+            with Session() as session:
+                attendee_count = int(session.execute(
+                    "SELECT reltuples AS count FROM pg_class WHERE relname = 'attendee'").scalar())
```
Comment (kitsuta) [inline uber/config.py]: Whoa, is this our first ever instance of pure SQL inside the codebase? 😮

Comment (RobRuana) [reply uber/config.py]: Maybe??? It will break on any non-Postgres database 😅

Comment (kitsuta) [reply uber/config.py]: I think realistically we don't have a lot of potential events going "I would love this, but it HAS to use MySQL" anyway :P

Comment (RobRuana) [reply uber/config.py]: Meh, they could just set `BADGES_SOLD_ESTIMATE_ENABLED = False` anyway!

**Pull request merged**